identity-account/src/events/command.rs

```diff
@@ -21,20 +23,23 @@ use crate::identity::TinyService;
 use crate::storage::Storage;
 use crate::types::Generation;
 use crate::types::KeyLocation;
+use crate::types::MethodSecret;
 
 // Supported authentication method types.
 const AUTH_TYPES: &[MethodType] = &[MethodType::Ed25519VerificationKey2018];
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug)]
 pub enum Command {
   CreateIdentity {
     network: Option<String>,
+    secret_key: Option<SecretKey>,
```

Comment:
Can we also make this a `MethodSecret`?

Reply:
No, we don't have to. Creating an identity from a KeyCollection does not need to be allowed according to @JelleMillenaar. It would also be difficult to add this, since the KeyCollection takes a parameter to create it. So when the user does not give a KeyCollection, it's unclear what the parameter should be.

Reply:
Right, we're unlikely to ever use `KeyCollection` for identity creation, I was just thinking in terms of forward compatibility with any other methods we might add and for consistency, since extending the `MethodSecret` enum would be non-breaking.